### ISSUE TITLE
Date Pipe fails at run-time after AOT compilation

### ISSUE THREAD

**Issue by yepuv1** [angular/angular#11896]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior** 
When compiled with ngc, the Date Pipe fails at runtime. Here is the stack trace
build.js:2 TypeError: Cannot convert undefined or null to object
    at new DateTimeFormat (native)
    at _e (http://localhost:3000/dist/build.js:1:21540)
    at Object.yMMMd (http://localhost:3000/dist/build.js:1:22017)
    at Oe (http://localhost:3000/dist/build.js:1:22086)
    at Function.t.format (http://localhost:3000/dist/build.js:6:3734)
    at t.transform (http://localhost:3000/dist/build.js:6:4065)
    at http://localhost:3000/dist/build.js:1:9119
    at e.detectChangesInternal (http://localhost:3000/dist/build.js:7:17240)
    at e.t.detectChanges (http://localhost:3000/dist/build.js:5:5356)
    at e.t.detectViewChildrenChanges (http://localhost:3000/dist/build.js:5:5904)

**Expected behavior**
Date pipe must work in AOT as it worked in JIT compilation.

**Reproduction of the problem**

Compiling the code below with ngc caused the error:

` import { Component } from '@angular/core';

@Component({
  selector: 'my-app',
  template: `<div>
    <p>Today is {{today | date}}</p>
    <p>Or if you prefer, {{today | date:'fullDate'}}</p>
    <p>The time is {{today | date:'jmZ'}}</p>
  </div>`
})
export class AppComponent {
    today:number = `Date.now();`
}
`

**Please tell us about your environment:**

Windows 10, VSCode, npm, lite, tsc 2.0.2

* **Angular version:** 2.0.X 
2.0.0

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ] 

 Chrome Version 53.0.2785.116 m (64-bit)
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
 ES5
* **Node (for AoT issues):** `node --version` =   
v6.2.2

**Comment by ericmartinezr** [comment]
dup of https://github.com/angular/angular/issues/11643

**Comment by yepuv1** [comment]
using option --locale=en-US , resolved the issue.
Thanks closing the issue

**Comment by vicb** [comment]
should be fixed in the latest release already (2.0.1)
